src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -340,44 +373,62 @@ protected override SupervisorStrategy SupervisorStrategy()
                 var directive = Directive.Stop;
 
                 ex.Match()
-                    .With<InvalidAddressAssociationException>(ia =>
+                    .With<InvalidAssociation>(ia =>
                     {
-                        log.Warning("Tried to associate with unreachable remote address [{0}]. Address is now gated for {1} ms, all messages to this address will be delivered to dead letters. Reason: [{2}]",
-                                 ia.RemoteAddress, settings.RetryGateClosedFor.TotalMilliseconds, ia.Message);
-                        endpoints.MarkAsFailed(Sender, Deadline.Now + settings.RetryGateClosedFor);
-                        AddressTerminatedTopic.Get(Context.System).Publish(new AddressTerminated(ia.RemoteAddress));
+                        KeepQuarantinedOr(ia.RemoteAddress, () =>
```

Comment:
Yes, minus the portion they implemented in a pull request 8-9 days ago: https://github.com/akka/akka/blob/master/akka-remote/src/main/scala/akka/remote/Remoting.scala#L445

I left a comment to go fill that in subsequently, because it touches a bunch of other files unrelated to this PR